File: packages/ios/Sources/Components/Toast/Toast.figma.swift
import SwiftUI
import Figma

struct KozmosToastBasicConnect: FigmaConnect {
    let component = KozmosToast.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8189"
    var variant = ["Content": "Basic"]

    @FigmaString("Title Text")
    var title: String = "Notification"

    @FigmaString("Description Text")
    var description: String = "Your changes have been saved."

    var body: some View {
        KozmosToast(
            isPresented: .constant(true),
            title: self.title,
            description: self.description,
            autoDismissAfter: nil
        )
    }
}

struct KozmosToastActionConnect: FigmaConnect {
    let component = KozmosToast.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8189"
    var variant = ["Content": "Action"]

    @FigmaString("Title Text")
    var title: String = "Notification"

    @FigmaString("Description Text")
    var description: String = "Your changes have been saved."

    @FigmaString("Action Text")
    var actionTitle: String = "Undo"

    var body: some View {
        KozmosToast(
            isPresented: .constant(true),
            title: self.title,
            description: self.description,
            actionTitle: self.actionTitle,
            autoDismissAfter: nil,
            onAction: {}
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/Toast/Toast.figma.kt
package com.kozmos.components.toast

import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.figma.code.connect.FigmaVariant

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8189")
@FigmaVariant("Content", "Basic")
class KozmosToastBasicConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Notification"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Your changes have been saved."

    @Composable
    fun ComponentExample() {
        KozmosToast(
            title = title,
            description = description,
            onDismiss = {}
        )
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8189")
@FigmaVariant("Content", "Action")
class KozmosToastActionConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Notification"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Your changes have been saved."

    @FigmaProperty(FigmaType.Text, "Action Text")
    val actionText: String = "Undo"

    @Composable
    fun ComponentExample() {
        KozmosToast(
            title = title,
            description = description,
            actionText = actionText,
            onAction = {},
            onDismiss = {}
        )
    }
}
